packages/tupaia-web/src/api/queries/useDashboards.ts

```diff
@@ -4,15 +4,31 @@
  */
 import { useQuery } from 'react-query';
 import { get } from '../api';
-import { DashboardsResponse } from '../../types';
+import { DashboardName, DashboardsResponse, EntityCode, ProjectCode } from '../../types';
 
-export const useDashboards = (projectCode?: string, entityCode?: string) => {
-  return useQuery(
+// Returns all dashboards for a project and entity, and also the active dashboard
+export const useDashboards = (
+  projectCode?: ProjectCode,
+  entityCode?: EntityCode,
+  dashboardName?: DashboardName,
+) => {
+  const { data: dashboards = [], isLoading } = useQuery(
     ['dashboards', projectCode, entityCode],
     (): Promise<DashboardsResponse[]> =>
       get('dashboards', {
         params: { entityCode, projectCode },
       }),
     { enabled: !!entityCode && !!projectCode },
   );
+
+  let activeDashboard = null;
+
+  if (dashboards.length > 0) {
```

Comment:
```suggestion
  if (dashboards.length > 0 && dashboard) {
```

Reply:
There might be places that we just want the dashboards but there isn't a dashboard name in the url yet